### PR TITLE
finished function and test, but test not passing

### PR DESCRIPTION
**Pull request by bashbashbash**
Working on getting tests to pass - I believe the function has an error in converting string data to bytes

### CODE REVIEW

File: internal/block/block_test.go [hunk 1]
```diff
@@ -457,3 +457,63 @@ func TestMarshal(t *testing.T) {
 		})
 	}
 }
+
+func TestUmarshal(t *testing.T) {
```
Comment (kstola-bloxroute) [inline internal/block/block_test.go]: sp

File: internal/block/block_test.go [hunk 2]
```diff
@@ -457,3 +457,63 @@ func TestMarshal(t *testing.T) {
...
+	testJSONBlock := JSONBlock{
```
Comment (kstola2) [inline internal/block/block_test.go]: Can you call `block.Marshall` on a regular block and use that output as the `JSONBlock`? I just want to make sure it's consistent with that function.

File: internal/block/block_test.go [hunk 3]
```diff
@@ -457,3 +457,69 @@ func TestMarshal(t *testing.T) {
 		})
 	}
 }
+
+func TestUmarshal(t *testing.T) {
```
Comment (kstola2) [inline internal/block/block_test.go]: Just need this typo fixed

**Comment by bashbashbash** [comment]
Completed #407

**Pull request merged**